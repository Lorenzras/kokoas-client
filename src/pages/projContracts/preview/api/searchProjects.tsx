import { APPIDS, KintoneRecord } from '../../../../api/kintone';

export const searchProjects = async <
  T extends KeyOfProjectDetails,
>(search: string) => {

  return KintoneRecord.getRecords({
    app: APPIDS.constructionDetails,
    query: `${'constructionName' as T} like "${search}"`,
    fields: ['constructionName', '$id' ] as T[],
    totalCount: true,
  })
    .then(r => r.records as unknown as ProjectDetails.SavedData[]);
};