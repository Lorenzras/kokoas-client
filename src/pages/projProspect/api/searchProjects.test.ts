import { searchProjects } from './searchProjects';

test('searchProjects', async ()=>{

  const result = await searchProjects('Lorenz');

  expect(result).toMatchSnapshot();

});