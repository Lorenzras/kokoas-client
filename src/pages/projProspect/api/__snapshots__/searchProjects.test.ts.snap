// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchProjects 1`] = `
Object {
  "records": Array [
    Object {
      "constructionName": Object {
        "type": "SINGLE_LINE_TEXT",
        "value": "Lorenz Ras様邸 蓄電池",
      },
    },
    Object {
      "constructionName": Object {
        "type": "SINGLE_LINE_TEXT",
        "value": "Lorenz Ras様邸 設備交換工事",
      },
    },
  ],
  "totalCount": "2",
}
`;
